providers/implementations/signature/eddsa_sig.c

```diff
@@ -43,6 +43,30 @@ static int s390x_ed448_digestverify(const ECX_KEY *edkey,
 
 #endif /* S390X_EC_ASM */
 
+/* JM: assigning a numeric id to each EdDSA instance
+       gives us a convenient way to record the instance
+       inside PROV_EDDSA_CTX.
+       official NID assignments are recorded inside obj_mac.h.
+       we could also just use an enum instead. */
+
+#define NID_Ed25519_INSTANCE    NID_ED25519
```

Comment:
There doesnt seem to be a good reason to use NID here.. The pattern used in other files is to use a table lookup.. e.g.

````
#define KEM_OP_UNDEFINED   -1
#define KEM_OP_RSASVE       0

static const OSSL_ITEM rsakem_opname_id_map[] = {
    { KEM_OP_RSASVE, OSSL_KEM_PARAM_OPERATION_RSASVE },
};

static int name2id(const char *name, const OSSL_ITEM *map, size_t sz)
{
    size_t i;

    if (name == NULL)
        return -1;

    for (i = 0; i < sz; ++i) {
        if (OPENSSL_strcasecmp(map[i].ptr, name) == 0)
            return map[i].id;
    }
    return -1;
}

static int rsakem_opname2id(const char *name)
{
    return name2id(name, rsakem_opname_id_map, OSSL_NELEM(rsakem_opname_id_map));
}
````

Reply:
okay.  I will refactor.